### PR TITLE
[FLINK-32443][docs-zh] Translate "State Processor API" page into Chinese

### PR DESCRIPTION
**Pull request by fredia**
## What is the purpose of the change

Translate docs/content.zh/docs/libs/state_processor_api.md into Chinese

## Brief change log

Translate docs/content.zh/docs/libs/state_processor_api.md into Chinese

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by fredia** [comment]
@flinkbot run azure

File: docs/content.zh/docs/libs/state_processor_api.md [hunk 1]
```diff
@@ -145,13 +130,13 @@ DataStream<Integer> listState = savepoint.readListState<>(
     new MyCustomIntSerializer());
 ```
 
-### Keyed State
+### 分区状态 Keyed State
 
-[Keyed state]({{< ref "docs/dev/datastream/fault-tolerance/state" >}}#keyed-state), or partitioned state, is any state that is partitioned relative to a key.
-When reading a keyed state, users specify the operator id and a `KeyedStateReaderFunction<KeyType, OutputType>`.
+[Keyed state]({{< ref "docs/dev/datastream/fault-tolerance/state" >}}#keyed-state)，又叫分区状态(partitioned state)，是与 key 相对应的状态。
```
Comment (Zakelly) [inline docs/content.zh/docs/libs/state_processor_api.md:135]:
```suggestion
[Keyed state]({{< ref "docs/dev/datastream/fault-tolerance/state" >}}#keyed-state)，又叫分区状态(partitioned state)，是使用一个 key 进行分区的状态。
```

File: docs/content.zh/docs/libs/state_processor_api.md [hunk 2]
```diff
@@ -223,19 +208,16 @@ public class ReaderFunction extends KeyedStateReaderFunction<Integer, KeyedState
 }
 ```
 
-Along with reading registered state values, each key has access to a `Context` with metadata such as registered event time and processing time timers.
-
-**Note:** When using a `KeyedStateReaderFunction`, all state descriptors must be registered eagerly inside of open. Any attempt to call a `RuntimeContext#get*State` will result in a `RuntimeException`.
+除了读取注册的状态之外，每个 key 还可以访问包括 event time 和 processing time [计时器](https://nightlies.apache.org/flink/flink-docs-release-1.17/docs/concepts/time/)等元数据的 `Context`。
 
-### Window State
+**注意：** 当使用 `KeyedStateReaderFunction` 时，所有状态描述符必须在 open 函数中注册。 否则任何尝试调用 `RuntimeContext#get*State` 将导致 `RuntimeException`。
```
Comment (Zakelly) [inline docs/content.zh/docs/libs/state_processor_api.md:213]:
```suggestion
**注意：** 当使用 `KeyedStateReaderFunction` 时，所有状态描述符必须在 `open` 函数中注册。 否则任何尝试调用 `RuntimeContext#get*State` 将导致 `RuntimeException`。
```

File: docs/content.zh/docs/libs/state_processor_api.md [hunk 3]
```diff
@@ -321,22 +303,19 @@ savepoint
 
 ```
 
-Additionally, trigger state - from `CountTrigger`s or custom triggers - can be read using the method
-`Context#triggerState` inside the `WindowReaderFunction`.
+另外，可以通过`WindowReaderFunction`里的`Context#triggerState`方法读取`CountTrigger`或自定义触发器的状态。
```
Comment (Zakelly) [inline docs/content.zh/docs/libs/state_processor_api.md:306]: 这里是否需要带空格，英文前后

File: docs/content.zh/docs/libs/state_processor_api.md [hunk 4]
```diff
@@ -321,22 +303,19 @@ savepoint
...
 
-## Writing New Savepoints
+## 通过 State Processor API 写出状态
```
Comment (Zakelly) [inline docs/content.zh/docs/libs/state_processor_api.md:308]:
```suggestion
## 通过 State Processor API 写出新状态
```

**Comment by fredia** [comment]
@Zakelly Thanks for the detailed review, I have addressed them, please take another look! thanks!

File: docs/content.zh/docs/libs/state_processor_api.md [hunk 5]
```diff
@@ -133,9 +118,9 @@ DataStream<Tuple2<Integer, Integer>> broadcastState = savepoint.readBroadcastSta
     Types.INT);
 ```
 
-#### Using Custom Serializers
+#### 使用自定义序列化器
 
-Each of the operator state readers support using custom `TypeSerializers` if one was used to define the `StateDescriptor` that wrote out the state.
+如果在写出状态时 `StateDescriptor` 使用了自定义的 `TypeSerializer`，Operator state readers 也支持使用自定义的 `TypeSerializers`。
```
Comment (Zakelly) [inline docs/content.zh/docs/libs/state_processor_api.md:123]:
```suggestion
如果在写出状态时 `StateDescriptor` 使用了自定义的 `TypeSerializer`，Operator state 也支持使用自定义的 `TypeSerializers` 来读取。
```

File: docs/content.zh/docs/libs/state_processor_api.md [hunk 6]
```diff
@@ -145,13 +130,13 @@ DataStream<Integer> listState = savepoint.readListState<>(
     new MyCustomIntSerializer());
 ```
 
-### Keyed State
+### 分区状态
```
Comment (Zakelly) [inline docs/content.zh/docs/libs/state_processor_api.md:133]: 我其实想说...这个专有名词就不用翻译了吧

**Review by Zakelly — COMMENTED** [review]
Thanks for the update. Overall LGTM except for several minor comments.

**Review by Zakelly — APPROVED** [review]
LGTM +1. Thanks for the update.

**Comment by fredia** [comment]
Thanks for the review, merging.

**Pull request merged**